### ISSUE TITLE
use fastcluster for hierarchical clustering

### ISSUE THREAD

**Issue by epurdom** [epurdom/clusterExperiment#237]
http://danifold.net/fastcluster.html

Should probably be as simple as calling the hclust from this package rather than stats.

**Comment by drisso** [comment]
Just implemented this in the `feature/fastcluster` branch. 

It was indeed as easy as changing a few lines to use `fastcluster::hclust`.

Assuming it passes all checks, can I merge into develop?

**Comment by drisso** [comment]
And obviously it failed... :|

**Comment by drisso** [comment]
So apparently `fastcluster::hclust()` and `stats::hclust()` do not return the same exact value, because when running the vignette with the same parameters, with stats we get two clusters and with fast cluster only one. This causes an error in `getBestFeatures` which makes the build fail.

**Comment by drisso** [comment]
Perhaps we should wait on this since hclust() is not the main bottleneck.

**Comment by epurdom** [comment]
I agree we should wait. Perhaps they just have different defaults that we need to align…
